src/charm.py

```diff
@@ -233,6 +247,50 @@ def update_external_services(self) -> None:
 
         self.expose_external = self.model.config["expose-external"]
 
+    def update_tls_sans(self) -> None:
+        """Emits a certificate expiring event when sans in current certificates are out of date.
+
+        This can occur for a variety of reasons:
+        1. Node port has been toggled on
+        2. Node port has been toggled off
+        3. The public K8s IP has changed
+        """
+
+        for internal in [True, False]:
+            # if the certificate has already been requested, we do not want to re-request
+            # another one and lead to an infinite chain of certificate events.
+            if self.tls.is_set_waiting_for_cert_to_update(internal):
+                continue
+
+            current_sans = self.tls.get_current_sans(internal)
+            current_sans_ip = set(current_sans["sans_ips"]) if current_sans else set()
+            expected_sans_ip = (
+                set(self.tls.get_new_sans()["sans_ips"]) if current_sans else set()
+            )
+            sans_ip_changed = current_sans_ip ^ expected_sans_ip
+
+            if not sans_ip_changed:
+                continue
```

Comment:
**todo:** I don't **think** you want to continue here, am I wrong?

If you `continue` if the sans has NOT changed, then you're always going to be re-requesting certs all the time.
I would have thought you wanted to do:
```python
if not sans_ip_changed:
    return
```

Reply:
I could be being dense. But I am fairly certain this is correct. If I am not understanding your point please correct me!

The reason we continue is because the loop is looping on`for internal in [True, False]:`, where `internal` represents which certificate we look at . (Charmed MongoDB uses two certificates, one for the replica's communication with each other, and another for replicas communication with clients)

Its important that we check both certificates, even if the first one doesn't need to be update.

Reply:
Duh, you're in a loop. Forgive me, all good, no changes here.